### PR TITLE
Omit all but instruction_addr field when symbol address not found

### PR DESCRIPTION
**Pull request by kstenerud**
## Goal

Stack trace entries whose symbol address cannot be determined will now only have an `instruction_addr` field.

Previously, it would send 0 values for missing fields, which would confuse the backend.

```
        "threads": [
            {
                "backtrace": {
                    "contents": [
                        {
                            "instruction_addr": 6230833072
                            "object_addr": 4367945728,
                            "symbol_addr": 0,
                        },
...
```

## Testing

Manually tested using:

```c
#if TARGET_CPU_X86 || TARGET_CPU_X86_64
    asm("jmp 0x70000000");
#elif TARGET_CPU_ARM || TARGET_CPU_ARM64
	asm volatile("bx %0" : : "r" (0x70000000));
#else
    #error(Unhandled CPU architecture)
#endif
```

Manually checked that the correct report is generated:

```
        "threads": [
            {
                "backtrace": {
                    "contents": [
                        {
                            "instruction_addr": 6230833072
                        },
                        {
                            "instruction_addr": 4368117122,
                            "object_addr": 4367945728,
                            "object_name": "UIKit",
                            "symbol_addr": 4368117039,
                            "symbol_name": "-[UIApplication sendAction:to:from:forEvent:]"
                        },
...
```

Manually checked backend display.

### CODE REVIEW

**Comment by nickdowell** [comment]
Could the manual test code be used to create an E2E scenario?

**Comment by kstenerud** [comment]
Unfortunately no. I tried creating e2e tests but they wouldn't give consistent results.

**Comment by nickdowell** [comment]
LGTM, just holding off until after today's release...

**Review by nickdowell — APPROVED** [review]

**Review by kattrali — APPROVED** [review]

**Pull request merged**
